test/unit/auth/tenant-manager.spec.ts

```diff
@@ -0,0 +1,590 @@
+        .should.eventually.be.rejected.and.have.property('code', 'app/invalid-credential');
+    });
+
+    it('should resolve with a Tenant on success', () => {
+      // Stub getTenant to return expected result.
+      const stub = sinon.stub(AuthRequestHandler.prototype, 'getTenant')
+        .returns(Promise.resolve(serverResponse));
+      stubs.push(stub);
+      return tenantManager.getTenant(tenantId)
+        .then((result) => {
+          // Confirm underlying API called with expected parameters.
+          expect(stub).to.have.been.calledOnce.and.calledWith(tenantId);
+          // Confirm expected tenant returned.
+          expect(result).to.deep.equal(expectedTenant);
+        });
+    });
+
+    it('should throw an error when the backend returns an error', () => {
+      // Stub getTenant to throw a backend error.
+      const stub = sinon.stub(AuthRequestHandler.prototype, 'getTenant')
+        .returns(Promise.reject(expectedError));
+      stubs.push(stub);
+      return tenantManager.getTenant(tenantId)
+        .then((tenant) => {
+          throw new Error('Unexpected success');
+        }, (error) => {
+          // Confirm underlying API called with expected parameters.
+          expect(stub).to.have.been.calledOnce.and.calledWith(tenantId);
+          // Confirm expected error returned.
+          expect(error).to.equal(expectedError);
+        });
+    });
+  });
+
+  describe('listTenants()', () => {
+    const expectedError = new FirebaseAuthError(AuthClientErrorCode.INTERNAL_ERROR);
+    const pageToken = 'PAGE_TOKEN';
+    const maxResult = 500;
+    const listTenantsResponse: any = {
+      tenants : [
+        {name: 'projects/project_id/tenants/tenant_id1'},
+        {name: 'projects/project_id/tenants/tenant_id2'},
+      ],
+      nextPageToken: 'NEXT_PAGE_TOKEN',
+    };
+    const expectedResult: ListTenantsResult = {
+      tenants: [
+        new Tenant({name: 'projects/project_id/tenants/tenant_id1'}),
+        new Tenant({name: 'projects/project_id/tenants/tenant_id2'}),
+      ],
+      pageToken: 'NEXT_PAGE_TOKEN',
+    };
+    const emptyListTenantsResponse: any = {
+      tenants: [],
+    };
+    const emptyExpectedResult: any = {
+      tenants: [],
+    };
+    // Stubs used to simulate underlying API calls.
+    let stubs: sinon.SinonStub[] = [];
+
+    afterEach(() => {
+      _.forEach(stubs, (stub) => stub.restore());
+      stubs = [];
+    });
+
+    it('should be rejected given an invalid page token', () => {
+      const invalidToken = {};
+      return tenantManager.listTenants(undefined, invalidToken as any)
+        .then(() => {
+          throw new Error('Unexpected success');
+        })
+        .catch((error) => {
+          expect(error).to.have.property('code', 'auth/invalid-page-token');
+        });
+    });
+
+    it('should be rejected given an invalid max result', () => {
+      const invalidResults = 5000;
```

Comment:
I suppose 1000 is the allowed max. I'd mention that in the test description or change the assignment to something like:

```
const moreThanMax = 1000 + 1;
```